ci/checks/check_copyright.py

```diff
@@ -0,0 +1,169 @@
+
+    Args:
+    repo_root - Root folder for git
+    """
+    os.chdir(repo_root)
+    branch = check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"])
+    branch = branch.replace('\n', '')
+    filelist = check_output(["git", "ls-tree", "-r", branch, "--name-only"])
+    tracked_files = list()
+    for tracked_file in filelist.splitlines():
+        tracked_files.append(os.path.join(repo_root, tracked_file))
+    return tracked_files
+
+
+def filter_files(files, regex):
+    """
+    Filter a list of file names by removing entries that match
+    the regular expression passed in.
+
+    Args:
+    files - List of file names
+    regex - Regular expression for filtering files
+    """
+    filtered_files = []
+    for fname in files:
+        if (not os.path.isfile(fname)):
+            continue
+        elif (not re.search(regex, fname)):
+            filtered_files.append(fname)
+    return filtered_files
+
+
+def add_copyright(f):
+    """
+    Add copyright header to the file passed in.
+
+    Args:
+    f - Path to file
+    """
+    extension = os.path.splitext(f)[1]
+    original_permissions = os.stat(f).st_mode
+    temp_f = f + ".copyright"
+
+    with open(temp_f, "w") as myfile:
+        copyright_text = cpp_copyright if extension in cpp_exts else other_copyright
+        with open(f, "r") as orig:
+            line = orig.readline()
+            if (line and '#!' in line):
+                # If file is a script and starts
+                # with #!, then keep that line
+                # as the first line and insert header
+                # after that.
+                myfile.write(line)
+                myfile.write("\n")
+                myfile.write(copyright_text)
+                line = orig.readline()
+            else:
+                myfile.write(copyright_text)
+            while line:
+                myfile.write(line)
+                line = orig.readline()
+    os.rename(temp_f, f)
+    # Resotre original file permissions
+    os.chmod(f, S_IMODE(original_permissions))
+
+
+def copyright_present(f):
+    """
+    Check if file already has copyright header.
+
+    Args:
+    f - Path to file
+    """
+    if 'Copyright (c)' in open(f).read():
+        return True
+    return False
+
+
+def parse_args():
+    """
+    Parse command line arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description="Add NVIDIA copyright headers to source files")
+    parser.add_argument('--add-header',
+                        help='Add copyright header to the files which do not have it',
+                        action='store_true')
+    args = parser.parse_args()
+    return args
+
+
+if __name__ == "__main__":
+    args = parse_args()
+
+    # Get list of files to check.
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+    root_dir = os.path.abspath(os.path.join(script_dir, "../../"))
+    files = get_tracked_files(root_dir)
+    files = filter_files(files, "LICENSE|README|data\/|docs\/")
```

Comment:
this string should be a raw string r"LICENSE|README|data\/|docs\/". Otherwise the `\/` character combination can be interpreted as an escape sequence (probably an invalid one) by different tools, editors etc